utils/deprecate_models.py

```diff
@@ -0,0 +1,472 @@
+"""
+
+import argparse
+import os
+from collections import defaultdict
+from pathlib import Path
+from typing import Optional, Tuple
+
+import requests
+from git import Repo
+from packaging import version
+
+from transformers import CONFIG_MAPPING, logging
+from transformers import __version__ as current_version
+
+
+REPO_PATH = Path(os.path.abspath(os.path.dirname(os.path.dirname(__file__))))
+repo = Repo(REPO_PATH)
+
+logger = logging.get_logger(__name__)
+
+
+def get_last_stable_minor_release():
+    # Get the last stable release of transformers
+    url = "https://pypi.org/pypi/transformers/json"
+    release_data = requests.get(url).json()
+
+    # Find the last stable release of of transformers (version below current version)
+    major_version, minor_version, patch_version, _ = current_version.split(".")
+    last_major_minor = f"{major_version}.{int(minor_version) - 1}"
+    last_stable_minor_releases = [
+        release for release in release_data["releases"] if release.startswith(last_major_minor)
+    ]
+    last_stable_release = sorted(last_stable_minor_releases, key=version.parse)[-1]
+
+    return last_stable_release
+
+
+def build_tip_message(last_stable_release):
+    return (
+        """
+    <Tip warning={true}>
+
+    This model is in maintenance mode only, we don't accept any new PRs changing its code.
+    """
+        + f"""If you run into any issues running this model, please reinstall the last version that supported this model: v{last_stable_release}.
+    You can do so by running the following command: `pip install -U transformers=={last_stable_release}`.
+
+    </Tip>"""
+    )
+
+
+def insert_tip_to_model_doc(model_doc_path, tip_message):
+    tip_message_lines = tip_message.split("\n")
+
+    with open(model_doc_path, "r") as f:
+        model_doc = f.read()
+
+    # Add the tip message to the model doc page directly underneath the title
+    lines = model_doc.split("\n")
+
+    new_model_lines = []
+    for line in lines:
+        if line.startswith("# "):
+            new_model_lines.append(line)
+            new_model_lines.extend(tip_message_lines)
+        else:
+            new_model_lines.append(line)
+
+    with open(model_doc_path, "w") as f:
+        f.write("\n".join(new_model_lines))
+
+
+def get_model_doc_path(model: str) -> Tuple[Optional[str], Optional[str]]:
+    model_doc_path = REPO_PATH / f"docs/source/en/model_doc/{model}.md"
+
+    if os.path.exists(model_doc_path):
+        return model_doc_path, model
+
+    # Try replacing _ with - in the model name
+    model_doc_path = REPO_PATH / f"docs/source/en/model_doc/{model.replace('_', '-')}.md"
+
+    if os.path.exists(model_doc_path):
+        return model_doc_path, model.replace("_", "-")
+
+    # Try replacing _ with "" in the model name
+    model_doc_path = REPO_PATH / f"docs/source/en/model_doc/{model.replace('_', '')}.md"
+
+    if os.path.exists(model_doc_path):
+        return model_doc_path, model.replace("_", "")
```

Comment:
This could probably be simplified with a for loop, where we 

> model_names_to_check = [ ... prepare all 3 variants ...]

then 

> for name in names_to_check:

Reply:
Good point - I've updated. Much cleaner :)